tests/GenerateScriptRegressionTests/CompareDirs.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Xunit.Abstractions;
+
+namespace GenerateScriptRegressionTests;
+
+internal class CompareDirs
+{
+    public static bool Compare(string pathA, string pathB, ITestOutputHelper outputHelper)
+    {
+
+        DirectoryInfo dir1 = new DirectoryInfo(pathA);
+        DirectoryInfo dir2 = new DirectoryInfo(pathB);
+
+        List<FileInfo> list1 = dir1.GetFiles("*.*", SearchOption.AllDirectories).ToList();
+        List<FileInfo> list2 = dir2.GetFiles("*.*", SearchOption.AllDirectories).ToList();
+
+        FileCompare myFileCompare = new FileCompare();
+
+        bool areIdentical = list1.SequenceEqual(list2, myFileCompare);
+
+        if (areIdentical == true)
+        {
+            for(int i = 0; i<list1.Count();i++)
+            {
+                areIdentical &= FileCompare(list1[i].FullName, list2[i].FullName);
+                outputHelper.WriteLine(string.Format("File {0} is modified!",list1[i].FullName));
+            }
+        }
+        return areIdentical;
+    }
+
+    private static bool FileCompare(string file1, string file2)
```

Comment:
In the primary source build tests we have utilize git to diff files as it provides a better UX for indicating the differences.  Consider utilizing it here - ExecuteHelper.ExecuteProcess("git", $"diff --no-index {file1Path} {file2Path}", outputHelper);

Reply:
I tried the git diff command. But will got many noise such as "warning: CRLF will be replaced by LF ..." and "The file will have its original line endings in your working directory".
I also tried the switch --ignore-cr-at-eol but it seems not work.

Reply:
I opened https://github.com/dotnet/source-build/issues/3500 to address this.

Reply:
Got it. Then I could wait until the issue fixed and switch to "git diff" for comparison.

Reply:
FYI https://github.com/dotnet/sdk/issues/33289 was resolved to you should be able to switch to git diff now.